### PR TITLE
Update `Get started` button to `Finish setup` when a user has draft data saved in the VBA flow

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
The button in the Workspace card page should change to Finish setup when a user has entered data into the form and closes the modal.

### Fixed Issues
$ Expensify/App#4892

### Tests
1. Login to an account with a Workspace and without an Expensify Card enabled or bank account setup.
2. Navigate to `Settings > Select Workspace > Get Started`.
3. Start filling out the form.
4. Close the form and notice that the button changed to `Finish setup`.

### QA Steps
Steps above.

### Tested On
- [X] Web
- [ ]  Mobile Web
- [X] Desktop
- [ ] iOS
- [ ]  Android

#### Web

https://user-images.githubusercontent.com/22219519/135545264-d7c246cc-2509-4f4d-b4de-ea5a475aab83.mov

#### Desktop

https://user-images.githubusercontent.com/22219519/135545274-d550ea37-0d8e-4575-9c22-d6fe846b043e.mov

#### Android

### CODE REVIEW

**Comment by AndrewGable** [comment]
Placing on hold, can you ping me when it's ready for merge and review? Thanks!

**Comment by luacmartins** [comment]
No longer on hold! @AndrewGable all yours for review!

**Review by AndrewGable — APPROVED** [review]

File: src/pages/workspace/WorkspaceCardPage.js [hunk 1]
```diff
@@ -66,148 +75,179 @@ const defaultProps = {
...
+        }
+        const buttonText = this.getButtonText();
+        this.setState({buttonText});
+        if (buttonText === this.props.translate('workspace.card.finishSetup')) {
+            this.openBankSetupModal();
+        }
+    }
+
+    shouldComponentUpdate(nextProps) {
+        if (this.props.isFocused || nextProps.isFocused) {
```
Comment (Jag96) [inline src/pages/workspace/WorkspaceCardPage.js:104]: nab - 1 line return

**Review by Jag96 — COMMENTED** [review]
One question and 1 nab, lgtm

File: src/libs/actions/BankAccounts.js [hunk 2]
```diff
@@ -338,13 +338,7 @@ function fetchUserWallet() {
 function fetchFreePlanVerifiedBankAccount(stepToOpen) {
     // We are using set here since we will rely on data from the server (not local data) to populate the VBA flow
     // and determine which step to navigate to.
-    Onyx.set(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {
-        loading: true,
-        error: '',
-
-        // We temporarily keep the achData state to prevent UI changes while fetching.
-        achData: {state: lodashGet(reimbursementAccountInSetup, 'state', '')},
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/libs/actions/BankAccounts.js:346]: We use the draft to derive `buttonText` in addition to `achData` and we are only updating `buttonText` when we exit the flow.

File: src/pages/workspace/WorkspaceCardPage.js (hunk 1 above)
Comment (luacmartins) [reply src/pages/workspace/WorkspaceCardPage.js:104]: Good one! Maybe next time 😉

**Comment by isagoico** [comment]
@luacmartins Hello! I think we're blocked from testing this. When creating an account with gmail it asks us to add a new domain with a work email. I think this might need to be internal QA.

**Comment by luacmartins** [comment]
@isagoico I updated the testing instructions. Could you please try again?

**Pull request merged**